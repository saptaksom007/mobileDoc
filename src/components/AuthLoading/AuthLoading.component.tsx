import React from 'react'
import { connect } from 'react-redux'
import { lt, pathOr, length, pipe } from 'ramda'
import { NavigationScreenProp } from 'react-navigation'

import { Loader } from 'components/Loader/Loader.component'
import { filterOldAppointment } from 'api/appointments'

interface Props {
  navigation?: NavigationScreenProp<any>
  logged?: boolean
  appointed?: boolean
}

const isLoggedIn = pathOr(false, ['isLoggedIn'])

const hasAppointment = pipe(
  pathOr([], ['rawList']),
  filterOldAppointment,
  length,
  lt(0),
)

export class AuthLoading extends React.Component<Props> {
  static navigationOptions = {
    headerMode: 'none',
  }

  componentDidMount() {
    this.navigate()
  }

  navigate = () => {
    const { appointed, logged, navigation } = this.props
    if (navigation) {
      if (appointed && !logged) {
        navigation.navigate('Appointment')
      } else if (logged) {
        navigation!.navigate('App')
      } else {
        navigation!.navigate('Auth')
      }
    }
  }

  componentDidUpdate(prevProps: Props) {
    if (prevProps.logged !== this.props.logged && this.props.logged === false) {
      if (this.props.appointed) {
        this.props.navigation!.navigate('Appointment')
      } else {
        this.props.navigation!.navigate('Auth')
      }
    }
  }

  render() {
    return <Loader info='AuthLoading' withLogo />
  }
}

export default connect(({ login, appointmentslist }: any) => ({
  logged: isLoggedIn(login),
  appointed: hasAppointment(appointmentslist),
}))(AuthLoading)
