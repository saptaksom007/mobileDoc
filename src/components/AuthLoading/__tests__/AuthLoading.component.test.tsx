import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { AuthLoading } from 'components/AuthLoading/AuthLoading.component'

jest.mock('components/Loader/Loader.component', () => ({ Loader: 'Loader' }))

it('<AuthLoading/> render correctly', () => {
  const tree = renderer.create(<AuthLoading />).toJSON()
  expect(tree).toMatchSnapshot()
})
