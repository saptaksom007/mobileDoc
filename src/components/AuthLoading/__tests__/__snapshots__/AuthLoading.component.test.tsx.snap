// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthLoading/> render correctly 1`] = `
<Loader
  info="AuthLoading"
  withLogo={true}
/>
`;
